src/ray/raylet/node_manager.cc

```diff
@@ -1199,19 +1199,26 @@ void NodeManager::ProcessRegisterClientRequestMessage(
       std::make_shared<Worker>(job_id, worker_id, language, worker_type,
                                worker_ip_address, client, client_call_manager_));
 
-  auto send_reply_callback = [this, client](Status status, int assigned_port) {
+  auto send_reply_callback = [this, client, job_id](Status status, int assigned_port) {
     flatbuffers::FlatBufferBuilder fbb;
     std::vector<std::string> system_config_keys;
     std::vector<std::string> system_config_values;
     for (auto kv : initial_config_.raylet_config) {
       system_config_keys.push_back(kv.first);
       system_config_values.push_back(kv.second);
     }
+
+    std::string serialized_job_config;
+    auto job_config = worker_pool_.GetJobConfig(job_id);
```

Comment:
This can be a reference right? 

```
const auto &job_config = 
```

Reply:
Thanks, but `WorkerPool::GetJobConfig` just reply `boost::optional<const rpc::JobConfig &>` rather than `const boost::optional<const rpc::JobConfig &> &`, so it could not use `const auto &job_config =` here.